### ISSUE TITLE
[QUESTION] Athena MSK query help

### ISSUE THREAD

**Issue by davidshtian** [awslabs/aws-athena-query-federation#899]
I've tried the latest released Athena MSK Integration feature, and using the official guidance from console to create the Lambda application. And after deployment (version 2022.50.1), I could not execute any queries, throwing lots of different kinds of errors and I went to check the CloudWatch logs, no logs found but messages showed:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder". | SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

Not sure this anything is missing, or it seems to be a logging dependency issue? 

Could anyone help on this? Thanks~

**Comment by henrymai** [comment]
@davidshtian

2022.50.2 has been deployed to fix the logging issue, but we still need more information about the errors that you've seen outside of the logging issue to help you further.

**Comment by davidshtian** [comment]
> @davidshtian
> 
> 2022.50.2 has been deployed to fix the logging issue, but we still need more information about the errors that you've seen outside of the logging issue to help you further.

Thanks for the fix~ The logging works fine, while there're still some other issues when query the data.

On the console, it throws 'Network Failure' error. And using the query editor to query the topic, it throws "SCHEMA_NOT_FOUND: line 2:6: Schema 'xxx' does not exist", as I'm not sure which schema name to be used. 

Suppose I've created Glue registry named _'athena-msk-registry'_, and the scheme in it called _'athena-msk-schema'_, and within the schema:

```
{
  "tableName": "orders",
  "schemaName": "customer_schema",
  "topicName": "orders",
  "message": {
...
```

And the data is mapped within _'default'_ database under MSK data source _'msk'_, so not sure what all these names's relationship and how to specify these names when querying the data?

```
select * from <data source name>.<schema name>.<table name>
```

I've tried Glue registry name, Glue schema name and schemaName as schema name, all responded with "SCHEMA_NOT_FOUND". Also tried default as schema name, and it responded with "TABLE_NOT_FOUND" instead. Hence, I not confused how to query the data within MSK? Or any special steps need to be taken on MSK side? Thanks~

Not seeing too much information from CloudWatch logs: 
```
2022-12-19 07:30:33  INFO  AbstractLogin:61 - Successfully logged in.
2022-12-19 07:30:34  INFO  AppInfoParser:119 - Kafka version: 3.3.1
2022-12-19 07:30:34  INFO  AppInfoParser:120 - Kafka commitId: e23c59d00e687ff5
2022-12-19 07:30:34  INFO  AppInfoParser:121 - Kafka startTimeMs: 1671435034098
START RequestId: 7aa8a7ba-4c0d-4d37-ba6a-58b2cae12c37 Version: $LATEST
2022-12-19 07:30:35 7aa8a7ba-4c0d-4d37-ba6a-58b2cae12c37 INFO  BaseAllocator:58 - Debug mode disabled.
2022-12-19 07:30:35 7aa8a7ba-4c0d-4d37-ba6a-58b2cae12c37 INFO  DefaultAllocationManagerOption:97 - allocation manager type not specified, using netty as the default type
2022-12-19 07:30:35 7aa8a7ba-4c0d-4d37-ba6a-58b2cae12c37 INFO  CheckAllocator:73 - Using DefaultAllocationManager at memory/DefaultAllocationManagerFactory.class
END RequestId: 7aa8a7ba-4c0d-4d37-ba6a-58b2cae12c37
```

**Comment by davidshtian** [comment]
Or if there is any detailed step-by-step guidance for Athena and MSK? Like easily, take MSK Serverless as a quick start example? Thanks~

**Comment by henrymai** [comment]
@davidshtian 

Have you seen this blog post? https://aws.amazon.com/blogs/big-data/analyze-real-time-streaming-data-in-amazon-msk-with-amazon-athena/

**Comment by davidshtian** [comment]
> @davidshtian
> 
> Have you seen this blog post? https://aws.amazon.com/blogs/big-data/analyze-real-time-streaming-data-in-amazon-msk-with-amazon-athena/

Yes, I've already seen it~ Almost the same process.

**Comment by henrymai** [comment]
In regard to the logs not showing much information, you will have to enable debug logging by setting this environment variable: `ATHENA_FEDERATION_SDK_LOG_LEVEL` to `debug` on your MSK Connector Lambda.

Can you also provide us with the queries you are trying to run?

Once the work week begins, I'll check with a team member that has more experience with the MSK setup and connector to chime in.

**Comment by davidshtian** [comment]
> In regard to the logs not showing much information, you will have to enable debug logging by setting this environment variable: `ATHENA_FEDERATION_SDK_LOG_LEVEL` to `debug` on your MSK Connector Lambda.
> 
> Can you also provide us with the queries you are trying to run?
> 
> Once the work week begins, I'll check with a team member that has more experience with the MSK setup and connector to chime in.
> 
> Another thing to check in the mean time is if your topic name in the glue schema that you've defined is the same as the topic name that you've defined in MSK (they need to be exactly the same including casing).

Thanks a lot~ I'lee try enabling debug loggin.

**Comment by henrymai** [comment]
This documentation might also help: https://docs.aws.amazon.com/athena/latest/ug/connectors-msk.html#connectors-msk-parameters

Make sure that you have all the parameters (environment variables on your Lambda) set up correctly.

**Comment by davidshtian** [comment]
@henrymai Thanks for your kindly help~ Finally, I've found out that _'Network Failure'_ error is due to Lambda VPC setting, I've attached public subnets (default route to IGW not NATGW) to this Lambda function, and cannot connect to Glue service, as the Lambda docs shown [here](https://docs.aws.amazon.com/lambda/latest/dg/configuration-vpc.html):

> To access private resources, connect your function to private subnets. If your function needs internet access, use [network address translation (NAT)](https://docs.aws.amazon.com/lambda/latest/dg/configuration-vpc.html#vpc-internet). Connecting a function to a public subnet doesn't give it internet access or a public IP address.

Then I add Glue VPC endpoint, and the network issue is gone, and then other issues are also fixed... Here, maybe it's better to add a Glue connection error in logs for debugging, otherwise it's hard to see what's happening.

After that, I could successfully query the data, and I've tested some different schema and table names:

```
select * from "<data source name>"."default"."<glue schema name>" : Good
select * from "default"."<glue schema name>" : Good

select * from "<data source name>"."default"."<tableName>" : Bad
select * from "default"."<tableName>" : Bad

select * from "<data source name>"."<schemaName>"."<glue schema name>" : Good
select * from "<schemaName>"."<glue schema name>" : Good

select * from "<data source name>"."<schemaName>"."<tableName>" : Bad
select * from "<schemaName>"."<tableName>" : Bad
```

It looks like that the schema name could be _'default'_ or schemaName within Glue schema, and the table name is Glue schema name not the tableName (or topicName) within Glue schema. However, Athena console shows the name as tableName (or topicName), meaning that the 'Preview Table' button will not generate the right schema and table name and thus throwing errors.

Not sure about the configuration in this [blog](https://aws.amazon.com/cn/blogs/big-data/analyze-real-time-streaming-data-in-amazon-msk-with-amazon-athena/), but it seems that creating the Glue schema with the name _'orders'_, otherwise it will not be queried successfully, and also if mismatch the naming (Glue schema not equal table name), Athena table details page will throw the error below, as it seeks the Glue schema with the table name.

```
Error loading table
Failed to invoke lambda function due to com.amazonaws.services.lambda.invoke.LambdaFunctionException: Schema is not found. RegistryName: null, SchemaName: null, SchemaArn: arn:aws:glue:us-east-1:xxxx:schema/athena-msk-registry/orders (Service: AWSGlue; Status Code: 400; Error Code: EntityNotFoundException; Request ID: xxxx; Proxy: null)
```

**Comment by davidshtian** [comment]
Just have another round of tests, the schema name could be _any_ string, not affecting the query, and it seems only topicName matters in Athena (not diving to source code yet).

**Comment by henrymai** [comment]
@davidshtian the fixed connector version is live now (2022.51.2)
